### PR TITLE
Pcm 919 typescript migration

### PR DESCRIPTION
**Pull request by djhouseknecht**
I still have to fix the unit tests, but the converting to TS is complete. Feel free to take a look and offer any feedback. I will let everyone know once it is stable and ready to be merged.

### CODE REVIEW

File: src/diagnostics/AdvancedCameraTest.ts [hunk 1]
```diff
@@ -26,7 +39,11 @@ export default class AdvancedCameraTest extends TestSuite {
     this.addTest(new CameraResolutionTest([[3840, 2160]], options));
     this.addTest(new CameraResolutionTest([[4096, 2160]], options));
   }
-  start () {
+
+  /**
+   * Start the test
+   */
+  public start (): Promise<any> {
```
Comment (xdumaine) [inline src/diagnostics/AdvancedCameraTest.ts]: I wonder if we could make this type more prescriptive. Results should all look similar, I imagine.

Comment (djhouseknecht) [reply src/diagnostics/AdvancedCameraTest.ts]: Ahh, you are right. I am already passing back a typed object. The type doesn't have an interface, but we know what it is. I will make sure to change this before my final PR.

Comment (djhouseknecht) [reply src/diagnostics/AdvancedCameraTest.ts]: I updated this return type in `AdvandedCameraTest`. I see what you are saying, though. I misunderstood your initial comment. I added a ticket in to standardize the return types across the tests. Right now, they return fairly different results per test. I definitely agree that they should all return a similar type.

File: package.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
   "version": "10.2.1",
```
Comment (xdumaine) [inline package.json]: Let's do a major version bump with these changes.

File: src/diagnostics/AdvancedCameraTest.ts [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
+  public name: string;
+
+  private promise: Promise<any>;
+  private deferred: { resolve: (...args: any[]) => void, reject: (...args: any[]) => void };
+
+  constructor (options: ObjectLiteral) {
     super(...arguments);
     this.name = 'Advanced Video Test';
-    this.tests = [];
+    // this.tests = [];
```
[the remark this answers is not in the document]
Comment (djhouseknecht) [reply src/diagnostics/AdvancedCameraTest.ts]: It looks like I removed that in my latest commit.

File: package.json (hunk 2 above)
Comment (djhouseknecht) [reply package.json]: I bumped it.

Comment (djhouseknecht) [reply package.json]: Done.

**Comment by xdumaine** [comment]
@djhouseknecht ready for this to merge?

**Comment by djhouseknecht** [comment]
Yep! Sorry, I just wanted to make sure I got the versioning right.

**Pull request merged**